sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java

```diff
@@ -179,21 +247,25 @@ public TableClient createTable(String tableName) {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<TableClient> createTableWithResponse(String tableName, Duration timeout, Context context) {
-        return blockWithOptionalTimeout(createTableWithResponse(tableName, context), timeout);
+        final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+        long timeoutInMillis = setTimeout(timeout);
+        Callable<Response<TableClient>> callable = () -> createTableWithResponse(tableName, context);
+        Future<Response<TableClient>> future = scheduler.submit(callable);
+        scheduler.shutdown();
+        try {
+            return future.get(timeoutInMillis, TimeUnit.MILLISECONDS);
+        } catch (Exception ex) {
+            throw logger.logExceptionAsError((RuntimeException) TableUtils.mapThrowableToTableServiceException(ex));
+        }
     }
 
-    Mono<Response<TableClient>> createTableWithResponse(String tableName, Context context) {
+    Response<TableClient> createTableWithResponse(String tableName, Context context) {
         context = context == null ? Context.NONE : context;
```

Comment:
Attach tracing context when sending request to the service.

Reply:
[Ditto question here](https://github.com/Azure/azure-sdk-for-java/pull/33143#discussion_r1084409393)